src/System.Private.CoreLib/Common/System/SR.cs

```diff
@@ -57,7 +58,7 @@ private static string InternalGetResourceString(string key)
             if (key == null || key.Length == 0)
             {
                 Debug.Fail("SR::GetResourceString with null or empty key.  Bug in caller, or weird recursive loading problem?");
-                return key;
+                return key!;
```

Comment:
TODO Comment? Debug.Fail will eventually be annotated with `NoReturn`.

Reply:
> Debug.Fail will eventually be annotated with NoReturn.

Debug.Fail is conditional and will compile away in release builds, such that we'd still get to `return key;`.